bundles/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/PersistenceExtensions.java

```diff
@@ -67,52 +91,193 @@ public PersistenceExtensions(@Reference PersistenceServiceRegistry registry) {
      *         if the provided <code>serviceId</code> does not refer to an available
      *         {@link QueryablePersistenceService}
      */
+    @Deprecated
     public static @Nullable HistoricItem historicState(Item item, ZonedDateTime timestamp, String serviceId) {
+        LoggerFactory.getLogger(PersistenceExtensions.class).info(
+                "The historicState method has been deprecated and will be removed in a future version, use persistedState instead.");
+        return internalPersistedState(item, timestamp, serviceId);
+    }
+
+    /**
+     * Retrieves the persisted item for a given <code>item</code> at a certain point in time through the default
+     * persistence service.
+     *
+     * @param item the item for which to retrieve the persisted item
+     * @param timestamp the point in time for which the persisted item should be retrieved
+     * @return the historic item at the given point in time, or <code>null</code> if no persisted item could be found,
+     *         the default persistence service is not available or does not refer to a
+     *         {@link QueryablePersistenceService}
+     */
+    public static @Nullable HistoricItem persistedState(Item item, ZonedDateTime timestamp) {
+        return internalPersistedState(item, timestamp);
+    }
+
+    /**
+     * Retrieves the persisted item for a given <code>item</code> at a certain point in time through a
+     * {@link PersistenceService} identified by the <code>serviceId</code>.
+     *
+     * @param item the item for which to retrieve the persisted item
+     * @param timestamp the point in time for which the persisted item should be retrieved
+     * @param serviceId the name of the {@link PersistenceService} to use
+     * @return the persisted item at the given point in time, or <code>null</code> if no persisted item could be found
+     *         or if the provided <code>serviceId</code> does not refer to an available
+     *         {@link QueryablePersistenceService}
+     */
+    public static @Nullable HistoricItem persistedState(Item item, ZonedDateTime timestamp, String serviceId) {
+        return internalPersistedState(item, timestamp, serviceId);
+    }
+
+    private static @Nullable HistoricItem internalPersistedState(Item item, @Nullable ZonedDateTime timestamp) {
+        String serviceId = getDefaultServiceId();
+        return serviceId != null ? internalPersistedState(item, timestamp, serviceId) : null;
+    }
+
+    private static @Nullable HistoricItem internalPersistedState(Item item, @Nullable ZonedDateTime timestamp,
+            String serviceId) {
+        if (timestamp == null) {
+            return null;
+        }
         PersistenceService service = getService(serviceId);
-        if (service instanceof QueryablePersistenceService qService) {
+        if (service != null && service instanceof QueryablePersistenceService qService) {
```

Comment:
Why is that needed? If `service` is an instance of `QueryablePersistenceService` is is not `null` (also applies to other methods).

Reply:
Indeed, not required.